network/common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java

```diff
@@ -50,6 +55,8 @@
 
   private final Map<Long, RpcResponseCallback> outstandingRpcs;
 
+  private final Queue<StreamCallback> streamCallbacks;
```

Comment:
I see.  this is why the `synchronized` block is in `TransportClient#stream`?  would be good to add a comment explaining that (maybe even something brief in both places).